libraries/chain/kv_context_rocksdb.cpp

```diff
@@ -146,9 +148,10 @@ namespace eosio { namespace chain {
          CATCH_AND_EXIT_DB_FAILURE()
 
          if (kv) {
-            if (offset < kv->value.size())
-               memcpy(dest, kv->value.data() + offset, std::min((size_t)size, kv->value.size() - offset));
-            actual_size = kv->value.size();
+            EOS_ASSERT(kv->value.size() >= kv_payer_size, kv_rocksdb_bad_value_size_exception , "The size of value returned from RocksDB is less than payer's size");
```

Comment:
make this a static method that takes  kv->value and have it return the actual value size, and use it here and at least 2 other locations below.

Reply:
Thanks for the suggestion. Will do.